From 6ee9ce78365926d6492dd01c99f86891be7bac40 Mon Sep 17 00:00:00 2001
From: Andy Duan <fugang.duan@nxp.com>
Date: Mon, 25 Jun 2018 14:47:15 +0800
Subject: [PATCH 4002/5242] MLK-17362-02 arm: dts: imx7ulp-evk: add qualcomm
 Qca9377-3 bt wifi support

commit  7f75f88b3b53b524cc5ba98a91f3c20d092b4804 from
https://source.codeaurora.org/external/imx/linux-imx.git

Add qualcomm Qca9377-3 bt wifi support for i.MX7ULP B0 chip
EVK RevA3 board.

Signed-off-by: Fugang Duan <fugang.duan@nxp.com>
(cherry picked and merged from commit: f85c441d095602edcd39d79db8509618a2f2f515)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/boot/dts/imx7ulp-evk.dts |   31 ++++++++++++++++++-------------
 1 file changed, 18 insertions(+), 13 deletions(-)

diff --git a/arch/arm/boot/dts/imx7ulp-evk.dts b/arch/arm/boot/dts/imx7ulp-evk.dts
index eb1f8a2..4251948 100644
--- a/arch/arm/boot/dts/imx7ulp-evk.dts
+++ b/arch/arm/boot/dts/imx7ulp-evk.dts
@@ -80,6 +80,16 @@
 			enable-active-high;
 		};
 
+		reg_sd1_vmmc: sd1_regulator {
+			compatible = "regulator-fixed";
+			regulator-name = "WLAN_EN";
+			regulator-min-microvolt = <3300000>;
+			regulator-max-microvolt = <3300000>;
+			gpio = <&rpmsg_gpio0 14 GPIO_ACTIVE_HIGH>;
+			off-on-delay = <20000>;
+			startup-delay-us = <100>;
+			enable-active-high;
+		};
 	};
 
 	extcon_usb1: extcon_usb1 {
@@ -333,7 +343,7 @@
 		pinctrl_usdhc1: usdhc1grp {
 			fsl,pins = <
 				IMX7ULP_PAD_PTE3__SDHC1_CMD	0x43
-				IMX7ULP_PAD_PTE2__SDHC1_CLK	0x43
+				IMX7ULP_PAD_PTE2__SDHC1_CLK	0x10042
 				IMX7ULP_PAD_PTE1__SDHC1_D0	0x43
 				IMX7ULP_PAD_PTE0__SDHC1_D1	0x43
 				IMX7ULP_PAD_PTE5__SDHC1_D2	0x43
@@ -350,12 +360,6 @@
 			>;
 		};
 
-		pinctrl_wifi: wifigrp {
-			fsl,pins = <
-				IMX7ULP_PAD_PTE6__PTE6		0x20043	/* WL_REG_ON */
-			>;
-		};
-
 		pinctrl_dsi_hdmi: dsi_hdmi_grp {
 			fsl,pins = <
 				IMX7ULP_PAD_PTC18__PTC18	0x10003	/* DSI_HDMI_INT */
@@ -550,12 +554,13 @@
 
 &usdhc1 {
 	pinctrl-names = "default", "sleep";
-	pinctrl-0 = <&pinctrl_usdhc1 &pinctrl_wifi>;
-	pinctrl-1 = <&pinctrl_usdhc1 &pinctrl_wifi>;
-	non-removable;
-	keep-power-in-suspend;
-	cd-post;
+	pinctrl-0 = <&pinctrl_usdhc1>;
+	pinctrl-1 = <&pinctrl_usdhc1>;
+	bus-width = <4>;
+	no-1-8-v;
+	vmmc-supply = <&reg_sd1_vmmc>;
 	pm-ignore-notify;
-	wifi-host;
+	keep-power-in-suspend;
+	non-removable;
 	status = "okay";
 };
-- 
1.7.9.5

